src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -39,13 +39,18 @@ public class ClaimsTransformationMiddleware
             handler.RegisterAuthenticationHandler(context.GetAuthentication());
             try
             {
-                if (Options.Transformer != null)
+                var transform = Options.Transformer;
```

Comment:
The transformer is initialized too late: it must be created before line 38 is invoked, or `ClaimsTransformationHandler` won't work properly.